src/cli/completion.h

```diff
@@ -0,0 +1,41 @@
+// vim: ts=4 sts=4 sw=4 et
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <CLI/CLI.hpp>
+#include <fmt/core.h>
+#include <fmt/ranges.h>
+
+#include <uenv/env.h>
+
+#include "uenv.h"
+
+namespace uenv {
+
+void completion_help();
+
+struct completion_args {
+    std::string shell_description;
```

Comment:
Store the `CLI::App*` in this type, and add a constructor ftakes the `CLI::App*` as an argument, instead of storing it in the main settings.